connectors/http/http-base/src/main/java/io/camunda/connector/http/base/exception/ConnectorExceptionMapper.java

```diff
@@ -16,32 +16,27 @@
  */
 package io.camunda.connector.http.base.exception;
 
-import com.fasterxml.jackson.core.JsonProcessingException;
 import io.camunda.connector.api.error.ConnectorException;
-import io.camunda.connector.api.json.ConnectorsObjectMapperSupplier;
+import io.camunda.connector.api.error.ConnectorExceptionBuilder;
 import io.camunda.connector.http.base.model.HttpCommonResult;
-import io.camunda.connector.http.base.utils.JsonHelper;
-import java.io.Serial;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.Optional;
 
-public class HttpCommonResultException extends ConnectorException {
+public class ConnectorExceptionMapper {
 
-  @Serial private static final long serialVersionUID = 1L;
-
-  public HttpCommonResultException(HttpCommonResult result) {
-    super(
-        String.valueOf(result.status()),
-        Optional.ofNullable(result.body())
-            .map(
-                body -> {
-                  try {
-                    return JsonHelper.isJsonValid(body)
-                        ? ConnectorsObjectMapperSupplier.DEFAULT_MAPPER.writeValueAsString(body)
-                        : String.valueOf(body);
-                  } catch (JsonProcessingException e) {
-                    throw new RuntimeException(e);
-                  }
-                })
-            .orElse(result.reason()));
+  public static ConnectorException from(HttpCommonResult result) {
+    String status = String.valueOf(result.status());
+    String reason = Optional.ofNullable(result.reason()).orElse("[no reason]");
+    Map<String, Object> headers = result.headers();
+    Object body = result.body();
+    Map<String, Object> response = new HashMap<>();
+    response.put("headers", headers);
+    response.put("body", body);
+    return new ConnectorExceptionBuilder()
+        .errorCode(status)
+        .message(reason)
+        .errorVariables(Map.of("response", response))
```

Comment:
Curious whether this is necessary or already done in https://github.com/camunda/connectors/blob/5c1577352bf5c043dade881ecadae4d20e0e783c/connector-sdk/feel-wrapper/src/main/java/io/camunda/connector/feel/FeelEngineWrapperUtil.java#L23 ?

Reply:
This is different. The code you posted just wraps a connector's response with this "response" key. It has nothing to do with errorVariables though